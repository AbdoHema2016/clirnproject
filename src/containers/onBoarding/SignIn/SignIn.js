import React from 'react';
import {
  View,
  KeyboardAvoidingView,
  Platform,
  Keyboard,
  Alert,
} from 'react-native';
import Style from './style';
import {connect} from 'react-redux';
import {loginAction} from './redux/actions';
import {visitingProfileViaAction} from '../../dashBoard/Home/redux/actions';
import CLabel from '../../../components/cLabel';
import CImage from '../../../components/cImage';
import CText from '../../../components/cTextField';
import CPassword from '../../../components/cPassword';
import CButton from '../../../components/cButton';
import {Constants, HelperFunctions, Validations} from '../../../utilities';
import {translate} from '../../../Localization';

import navigationService from '../../../Navigation/NavigationService';
import dynamicLinks from '@react-native-firebase/dynamic-links';
import AsyncStorage, {AsyncConstants} from '../../../utilities/AsyncStorage';
import {createSignature, ifKeyAlreadyExists} from '../../Biomitric/Biometrics';
import {BiometricSignAction} from '../../Biomitric/redux/actions';
import {getHash} from '../otp/OTPDetect';
const {showErrorMessage} = HelperFunctions;
const {LOCAL_PATH, testIds} = Constants;
const {validateEmail, validatePassword} = Validations;
class SignIn extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      checkValidation: false,
      username: {value: '', empty: true, isValid: false, status: ''},
      password: {
        value: '',
        empty: true,
        isValid: false,
        status: '',
        hidden: true,
      },
      biometricEnabled: false,
    };
    navigationService.navigation = this.props.navigation;
  }

  doLogin = () => {
    const {visitingProfileVia, login} = this.props;
    const {username, password} = this.state;
    if (
      username.value === 'gauravrana7777@gmail.com' &&
      Platform.OS === 'android'
    ) {
      getHash((cb) => {
        Alert.alert(cb);
      });
    }
    this.setState({checkValidation: true}, () => {
      if (!password.isValid || !username.isValid) {
        return;
      }

      login(username.value, password.value);
      visitingProfileVia('SignIn');
    });
  };

  goToBenefits = () => {
    this.props.navigation.navigate('Benefits');
  };

  goToForgotenPassword = () => {
    this.props.navigation.navigate('ForgotPassword');
  };

  biometricLogin = async () => {
    const {biometricSign} = this.props;
    try {
      let email = await AsyncStorage.getItemFromStorage(
        AsyncConstants.USER_EMAIL,
      );
      let device_token = await AsyncStorage.getItemFromStorage(
        AsyncConstants.DEVICE_TOKEN,
      );
      let biometric_id = await AsyncStorage.getItemFromStorage(
        AsyncConstants.BIOMETRIC_ID,
      );
      let signature = await createSignature(
        `${translate('LOGIN_SCREEN_STRINGS.CONTINUE_TO_LABEL')} ${email}`,
        email.toLowerCase(),
      );

      if (!signature) {
        throw new Error();
      }
      biometricSign({email, signature, device_token, biometric_id});
    } catch {
      showErrorMessage(translate('STRINGS.SOMETHING_WENT_WRONG'));
    }
  };
  editLoginFields = (label, value) => {
    const {username, password} = this.state;

    if (label === 'username') {
      this.setState({
        username: {
          value: value,
          status: 'Focused',
          empty: value.trim() === '' ? true : false,
          isValid: validateEmail(value.trim()),
        },
        password: {
          ...password,
          status: !password.value ? 'Active' : '',
        },
      });
      return;
    }
    this.setState({
      password: {
        ...password,
        value: value,
        status: 'Focused',
        empty: value.trim() === '' ? true : false,
        isValid: validatePassword(value.trim()),
      },
      username: {
        ...username,
        status: !username.value ? 'Active' : '',
      },
    });
    return;
  };

  focusPasswordField = () => {
    this.password.focus();
  };

  checkEmailError = () => {
    const {username} = this.state;

    if (!username.isValid && this.state.checkValidation) {
      return true;
    }
  };

  returnEmailError = () => {
    const {username} = this.state;

    if (username.empty) {
      return translate('LOGIN_SCREEN_STRINGS.ERROR_EMPTY_ID');
    }
    if (!username.isValid) {
      return translate('LOGIN_SCREEN_STRINGS.ERROR_INVALID_ID');
    }
  };

  checkPasswordError = () => {
    const {password} = this.state;
    const {checkValidation} = this.state;

    if (!password.isValid && checkValidation) {
      return true;
    }

    return false;
  };

  returnPasswordError = () => {
    const {password} = this.state;

    if (password.empty) {
      return translate('LOGIN_SCREEN_STRINGS.ERROR_EMPTY_PASSWORD');
    }
    if (!password.isValid) {
      return translate('LOGIN_SCREEN_STRINGS.ERROR_INVALID_PASSWORD');
    }
  };
  componentDidMount() {
    this.unsubscribe = dynamicLinks().onLink(this.handleDynamicLink);
    this.checkBiometric();
  }
  checkBiometric = async () => {
    let biometricEnabledCheck = await AsyncStorage.getItemFromStorage(
      AsyncConstants.BIOMETRIC_ENABLED,
    );
    if (biometricEnabledCheck) {
      this.setState({biometricEnabled: true});
    }
  };
  handleDynamicLink = async (link) => {
    AsyncStorage.setItemInStorage(AsyncConstants.LINKING, link);
  };
  componentWillUnmount() {
    this.unsubscribe();
  }
  hidePassword = () => {
    const {password} = this.state;

    this.setState({
      password: {
        ...password,
        hidden: !password.hidden,
      },
    });
  };
  render() {
    const {
      signInState: {loading},
    } = this.props;
    const {username, password, biometricEnabled} = this.state;
    return (
      <View style={Style.container}>
        <KeyboardAvoidingView
          style={Style.keyboardAvoidingView}
          behavior={Platform.OS === 'ios' ? 'position' : 'height'}
          keyboardVerticalOffset={Style.keyboardVerticalOffset.top}>
          <View style={Style.innerContainer}>
            <View style={Style.appTitleContainer}>
              <CImage
                type={'local'}
                imagePath={LOCAL_PATH.SPLASH_ICON}
                resizeMode="center"
                imageStyle={Style.logo}
              />
            </View>

            <View style={Style.inputContainer}>
              <CLabel
                style={Style.inputLabel}
                text={translate('CONTACT_DETAILS_SCREEN_STRINGS.EMAIL_LABEL')}
              />
              <CText
                accessibilityLabel={testIds.emailSignIn}
                blurOnSubmit={false}
                onSubmitEditing={this.focusPasswordField}
                returnKeyType={'next'}
                placeHolderText={translate(
                  'CONTACT_DETAILS_SCREEN_STRINGS.EMAIL_LABEL',
                )}
                keyboardType={'email-address'}
                onChangeText={(val) => {
                  this.editLoginFields('username', val);
                }}
                state={username.status}
                error={this.checkEmailError()}
                errorMessage={this.returnEmailError()}
                autoCapitalize={'none'}
              />
              <CLabel
                style={Style.inputLabel}
                text={translate('LOGIN_SCREEN_STRINGS.PASSWORD_LABEL')}
              />
              <CPassword
                accessibilityLabel={testIds.passwordSignIn}
                blurOnSubmit={true}
                reference={(input) => {
                  this.password = input;
                }}
                editable={true}
                hidePasswordAction={this.hidePassword}
                hidePassword={password.hidden}
                placeHolderText={translate(
                  'LOGIN_SCREEN_STRINGS.PASSWORD_LABEL',
                )}
                state={password.status}
                customSuperContainer={Style.inputTextsupreContainerStyle}
                containerStyle={Style.inputTextStyle}
                returnKeyType={'done'}
                onChangeText={(val) => {
                  this.editLoginFields('password', val);
                }}
                onSubmitEditing={Keyboard.dismiss}
                error={this.checkPasswordError()}
                errorMessage={this.returnPasswordError()}
              />
              {ifKeyAlreadyExists && biometricEnabled && (
                <CLabel
                  onPress={this.biometricLogin}
                  style={Style.Biometric}
                  text={translate('LOGIN_SCREEN_STRINGS.LOGIN_TOUCH_FACE_ID')}
                />
              )}

              <CLabel
                accessibilityLabel={testIds.passwordForgotten}
                onPress={this.goToForgotenPassword}
                style={Style.forgotenPasswordLink}
                text={translate('LOGIN_SCREEN_STRINGS.PASSWORD_FORGOTTEN')}
              />
            </View>
          </View>
        </KeyboardAvoidingView>
        <View style={Style.buttonsContainer}>
          <CButton
            accessibilityLabel={testIds.signIn}
            text={translate('LOGIN_SCREEN_STRINGS.CONTINUE_LABEL')}
            textStyle={Style.whiteText}
            onPress={this.doLogin}
            loading={loading}
            buttonContainerStyle={Style.blackButton}
          />
          <CButton
            accessibilityLabel={testIds.continueToBenefits}
            text={translate('LOGIN_SCREEN_STRINGS.CREATE_ACCOUNT_LABEL')}
            textStyle={Style.blackText}
            marginTop={Style.whiteButton.marginTop}
            onPress={this.goToBenefits}
            backgroundColor={Style.whiteButton.backgroundColor}
            buttonContainerStyle={Style.whiteButton}
          />
        </View>
      </View>
    );
  }
}

const mapStateToProps = ({signIn: signInState}) => ({signInState});

const mapDispatchToProps = {
  login: loginAction,
  visitingProfileVia: visitingProfileViaAction,
  biometricSign: BiometricSignAction,
};

export default connect(mapStateToProps, mapDispatchToProps)(SignIn);
