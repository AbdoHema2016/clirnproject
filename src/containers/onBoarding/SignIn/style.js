import {StyleSheet, Platform} from 'react-native';
import {Layout} from '../../../utilities';

const {LAYOUT_CONSTRAINTS, FONTS, HEX_COLOR_CODES} = Layout;

const Style = StyleSheet.create({
  keyboardAvoidingView: {
    height:
      LAYOUT_CONSTRAINTS.SCREEN_HEIGHT > 800
        ? Platform.OS === 'ios'
          ? LAYOUT_CONSTRAINTS.SCREEN_HEIGHT -
            (LAYOUT_CONSTRAINTS.SCREEN_HEIGHT * 0.22 + 80)
          : LAYOUT_CONSTRAINTS.SCREEN_HEIGHT -
            (LAYOUT_CONSTRAINTS.SCREEN_HEIGHT * 0.22 + 160)
        : LAYOUT_CONSTRAINTS.SCREEN_HEIGHT - 240,
  },
  keyboardVerticalOffset: {
    top: Platform.OS === 'android' ? 44 : 80,
  },
  container: {
    flex: 1,
    alignItems: 'center',
    backgroundColor: HEX_COLOR_CODES.LIGHT_GRAY,
  },
  innerContainer: {
    alignItems: 'center',
  },
  scrollView: {
    flex: 1,
  },
  forgotenPasswordLink: {
    color: HEX_COLOR_CODES.BLACK,
    fontFamily: FONTS.SOFIA_MEDIUM,
    fontSize: 12,
    marginRight: 8,
    marginTop: 8,
    alignSelf: 'flex-end',
    textDecorationLine: 'underline',
  },
  Biometric: {
    color: HEX_COLOR_CODES.BLACK,
    fontFamily: FONTS.SOFIA_MEDIUM,
    fontSize: 12,
    marginRight: 8,
    marginTop: 8,
    alignSelf: 'flex-start',
  },
  appTitleContainer: {
    flexDirection: 'row',
    ...Platform.select({
      ios: {
        marginTop:
          LAYOUT_CONSTRAINTS.SCREEN_HEIGHT > 800
            ? LAYOUT_CONSTRAINTS.SCREEN_HEIGHT * 0.19
            : LAYOUT_CONSTRAINTS.SCREEN_HEIGHT * 0.14,
      },
      android: {
        marginTop:
          LAYOUT_CONSTRAINTS.SCREEN_HEIGHT > 800
            ? LAYOUT_CONSTRAINTS.SCREEN_HEIGHT * 0.17
            : LAYOUT_CONSTRAINTS.SCREEN_HEIGHT * 0.09,
      },
    }),
    width: 'auto',
    alignItems: 'center',
  },
  logo: {
    width: LAYOUT_CONSTRAINTS.SCREEN_WIDTH,
    height: LAYOUT_CONSTRAINTS.SCREEN_WIDTH * 0.3,
  },
  appTitle: {
    fontSize: 32,
    fontWeight: '600',
    marginLeft: 20.32,
    color: HEX_COLOR_CODES.BLACK,
    fontFamily: FONTS.SOFIA_MEDIUM,
  },
  aboutAppLabelContainer: {
    marginTop: 32,
    width: 'auto',
  },
  aboutAppLabel: {
    fontSize: 18,
    fontWeight: '600',
    color: HEX_COLOR_CODES.BLACK,
    textAlign: 'center',
    width: 220,
  },
  inputContainer: {
    width: LAYOUT_CONSTRAINTS.SCREEN_WIDTH - 24,
    marginTop: LAYOUT_CONSTRAINTS.SCREEN_HEIGHT > 800 ? 44 : 8,
    height: 'auto',
  },
  inputLabel: {
    height: 16,
    marginTop: 16,
    fontSize: 12,
    color: HEX_COLOR_CODES.BLACK,
    fontFamily: FONTS.SOFIA_REGULAR,
  },
  buttonsContainer: {
    position: 'absolute',
    ...Platform.select({
      ios: {
        bottom: LAYOUT_CONSTRAINTS.WINDOW_HEIGHT * 0.12,
      },
      android: {
        bottom: LAYOUT_CONSTRAINTS.WINDOW_HEIGHT * 0.11,
      },
    }),
    width: LAYOUT_CONSTRAINTS.WINDOW_WIDTH - 28,
    height: 124,
  },
  blackButton: {
    backgroundColor: HEX_COLOR_CODES.BLACK_BUTTON,
  },
  whiteButton: {
    backgroundColor: HEX_COLOR_CODES.WHITE,
    marginTop: 16,
  },
  whiteText: {
    color: HEX_COLOR_CODES.WHITE,
  },
  blackText: {
    color: HEX_COLOR_CODES.BLACK,
  },
});

export default Style;
