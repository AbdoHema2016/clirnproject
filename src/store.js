import {createStore, applyMiddleware} from 'redux';
import createSagaMiddleware from 'redux-saga';
import {createLogger} from 'redux-logger';
import rootReducer from './reducers';
import {rootSaga} from './sagas';
const sagaMiddleware = createSagaMiddleware();
const isDevelopmentEnv =
  process && process.env && process.env.NODE_ENV === 'development';
const isProductionEnv =
  process && process.env && process.env.NODE_ENV === 'production';
const logStoreInfo = false;
const middlewares = [sagaMiddleware];
function actionLogger({getState}) {
  return (next) => (action) => {
    let allowed = [
      'jumio',
      'notificationsReducer',
      'otp',
      'companies',
      'signIn',
      'updateContactDetailsReducer',
      'userProfile',
      'vaccinations',
    ];
    let state = getState();
    let loggedStates = Object.keys(state)
      .filter((key) => allowed.includes(key))
      .reduce((obj, key) => {
        return {
          ...obj,
          [key]: state[key],
        };
      }, {});
    console.info({actionType: action.type, loggedStates});
    const returnValue = next(action);
    return returnValue;
  };
}
if (isDevelopmentEnv && logStoreInfo) {
  middlewares.push(createLogger());
}
if (isProductionEnv) {
  middlewares.push(actionLogger);
}

const store = createStore(rootReducer, applyMiddleware(...middlewares));
sagaMiddleware.run(rootSaga);
export default store;
