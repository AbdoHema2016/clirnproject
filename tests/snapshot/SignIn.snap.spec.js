import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import store from '../../src/store';
import SignIn from '../../src/containers/onBoarding/SignIn/SignIn';
jest.mock('react-native-otp-verify', () => ({
  getHash: jest.fn().mockImplementation(() => Promise.reject()),
  getOtp: jest.fn(),
  removeListener: jest.fn(),
}));
describe('Snapshot screen', () => {
  const props = {
    navigation: {
      setOptions: jest.fn(),
    },
  };
  it('Snapshot Sign', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <SignIn {...props} />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
