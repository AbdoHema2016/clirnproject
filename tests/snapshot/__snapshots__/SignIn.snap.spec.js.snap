// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot screen Snapshot Sign 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F7F7F7",
      "flex": 1,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "height": 880.52,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "marginTop": 253.46,
            "width": "auto",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "height": 225,
                "width": 750,
              },
            ]
          }
        >
          <FastImageView
            resizeMode="center"
            source={
              Object {
                "cache": "immutable",
                "uri": undefined,
              }
            }
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "height": "auto",
            "marginTop": 44,
            "width": 726,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#2B342C",
                "fontFamily": "SofiaProRegular",
                "fontSize": 12,
                "height": 16,
                "marginTop": 16,
              },
            ]
          }
        >
          [missing "en.CONTACT_DETAILS_SCREEN_STRINGS.EMAIL_LABEL" translation]
        </Text>
        <View
          style={
            Array [
              Object {
                "height": "auto",
                "width": 724,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFFFFF",
                  "borderRadius": 4,
                  "height": 48,
                  "justifyContent": "center",
                  "marginTop": 4,
                  "width": "100%",
                },
                undefined,
                null,
              ]
            }
          >
            <TextInput
              accessibilityLabel="Email@1"
              accessible={true}
              allowFontScaling={true}
              autoCapitalize="none"
              blurOnSubmit={false}
              keyboardType="email-address"
              numberOfLines={1}
              onChangeText={[Function]}
              onSubmitEditing={[Function]}
              placeholder="[missing \\"en.CONTACT_DETAILS_SCREEN_STRINGS.EMAIL_LABEL\\" translation]"
              placeholderTextColor="#8D8C91"
              rejectResponderTermination={true}
              returnKeyType="next"
              style={
                Array [
                  Object {
                    "color": "#2B342C",
                    "fontSize": 18,
                    "fontWeight": "500",
                    "lineHeight": 20,
                    "paddingLeft": 19,
                    "paddingRight": 19,
                    "width": "100%",
                  },
                  Object {},
                ]
              }
              underlineColorAndroid="transparent"
            />
          </View>
        </View>
        <Text
          style={
            Array [
              Object {
                "color": "#2B342C",
                "fontFamily": "SofiaProRegular",
                "fontSize": 12,
                "height": 16,
                "marginTop": 16,
              },
            ]
          }
        >
          [missing "en.LOGIN_SCREEN_STRINGS.PASSWORD_LABEL" translation]
        </Text>
        <View
          style={
            Array [
              Object {
                "height": "auto",
                "width": 724,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFFFFF",
                  "borderRadius": 4,
                  "height": 48,
                  "justifyContent": "center",
                  "marginTop": 4,
                  "width": "100%",
                },
                undefined,
                Object {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <TextInput
              accessibilityLabel="Password@1"
              allowFontScaling={true}
              blurOnSubmit={false}
              editable={true}
              onChangeText={[Function]}
              onSubmitEditing={[Function]}
              placeholder="[missing \\"en.LOGIN_SCREEN_STRINGS.PASSWORD_LABEL\\" translation]"
              placeholderTextColor="#8D8C91"
              rejectResponderTermination={true}
              returnKeyType="done"
              secureTextEntry={true}
              style={
                Array [
                  Object {
                    "color": "#2B342C",
                    "fontSize": 18,
                    "fontWeight": "500",
                    "lineHeight": 20,
                    "paddingLeft": 19,
                    "paddingRight": 19,
                    "width": "100%",
                  },
                  Object {
                    "marginLeft": 0,
                    "width": "90%",
                  },
                ]
              }
              underlineColorAndroid="transparent"
            />
            <View
              accessibilityLabel="Eye"
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID="Eye"
            >
              <View
                style={
                  Array [
                    Object {
                      "overflow": "hidden",
                    },
                    Object {
                      "height": 25,
                      "marginRight": 18,
                      "width": 25,
                    },
                  ]
                }
              >
                <FastImageView
                  resizeMode="cover"
                  source={
                    Object {
                      "cache": "immutable",
                      "uri": undefined,
                    }
                  }
                  style={
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                />
              </View>
            </View>
          </View>
        </View>
        <Text
          accessibilityLabel="PasswordForgotten"
          onPress={[Function]}
          style={
            Array [
              Object {
                "alignSelf": "flex-end",
                "color": "#2B342C",
                "fontFamily": "SofiaPro-Medium",
                "fontSize": 12,
                "marginRight": 8,
                "marginTop": 8,
                "textDecorationLine": "underline",
              },
            ]
          }
        >
          [missing "en.LOGIN_SCREEN_STRINGS.PASSWORD_FORGOTTEN" translation]
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "bottom": 160.07999999999998,
        "height": 124,
        "position": "absolute",
        "width": 722,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 4,
            "height": 52,
            "justifyContent": "center",
            "shadowColor": "#DDDDDD",
            "shadowOffset": Object {
              "height": 8,
              "width": 0,
            },
            "shadowOpacity": 0.75,
            "shadowRadius": 4,
          },
          Object {
            "backgroundColor": "#565655",
          },
        ]
      }
    >
      <View
        accessibilityLabel="Continue@1"
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": undefined,
            "borderRadius": 4,
            "height": "100%",
            "justifyContent": "center",
            "opacity": 1,
            "width": 718,
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "fontFamily": "SofiaProRegular",
                "fontSize": 17,
                "textAlign": "center",
              },
              Object {
                "color": "#FFFFFF",
              },
            ]
          }
        >
          [missing "en.LOGIN_SCREEN_STRINGS.CONTINUE_LABEL" translation]
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 4,
            "height": 52,
            "justifyContent": "center",
            "shadowColor": "#DDDDDD",
            "shadowOffset": Object {
              "height": 8,
              "width": 0,
            },
            "shadowOpacity": 0.75,
            "shadowRadius": 4,
          },
          Object {
            "backgroundColor": "#FFFFFF",
            "marginTop": 16,
          },
        ]
      }
    >
      <View
        accessibilityLabel="ContinueToBenefits"
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderRadius": 4,
            "height": "100%",
            "justifyContent": "center",
            "opacity": 1,
            "width": 718,
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "fontFamily": "SofiaProRegular",
                "fontSize": 17,
                "textAlign": "center",
              },
              Object {
                "color": "#2B342C",
              },
            ]
          }
        >
          [missing "en.LOGIN_SCREEN_STRINGS.CREATE_ACCOUNT_LABEL" translation]
        </Text>
      </View>
    </View>
  </View>
</View>
`;
